woocommerce/woocommerce-ios
# Observe keyboard and reposition `Notice` in `DefaultNoticePresenter`

Closes #4942

## Description
It has been observed and reported that Notices presented using the `NoticePresenter` don’t reposition themselves, when keyboard is presented or dismissed. i.e. Notice gets hidden by the keyboard.

This change includes the code to make the NoticePresenter update/animate the position of of the Notice, as per keyboard’s presence.

## Testing
The following is a scenario in which Notice can be tested.
1. Go to the `Products` tab.
2. Tap on the product and tap on the description.
3. Change the description and tap `Done`. 
4. In the Product form screen, tap on `Save` to save the edited description.
5. The description will be saved and observe that `Product saved` Notice is being presented from the bottom.
5. Quickly tap on anyone of the Title or Description textfields, and observe that the Notice is moved up as the keyboard appears.

By making local and temporary code changes, I have also tested the case in which **the keyboard is already presented and a Notice appears** and Notice is presented above the keyboard.
I also made sure that **Notice is moved to the bottom when keyboard is dismissed**.


https://user-images.githubusercontent.com/524475/135725340-e8b8ea5c-132d-4fd5-a276-e882d855def9.mov


Update release notes:
